tests/utils/object_store/test_object_store.py

```diff
@@ -55,11 +90,13 @@ def object_store(request, monkeypatch: pytest.MonkeyPatch,
         os.makedirs(remote_dir)
         object_store_kwargs[request.param]['provider_kwargs']['key'] = remote_dir
         yield request.param(**object_store_kwargs[request.param])
+    elif request.param is SFTPObjectStore:
+        yield TestSFTPObjectStore("unused_test_hostname")
```

Comment:
Instead of mocking out the entire init with `TestSFTPObjectStore`, would it be possible to monkeypatch the Server class to return the mock server class, so we can still test the original init?